plugins/cassandra/cassandraembed.cpp

```diff
@@ -432,7 +439,7 @@ class CassandraStatementInfo : public CInterface
     {
         stop();
         futures.kill();
-        delete semaphore;
+        semaphore.clear();
```

Comment:
I have a higher opinion of compiler capabilities than you. But it's irrelevant, since the extra work in question is insignificant.